pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -2940,3 +2940,50 @@ TEST(BasicEndToEndTest, testPreventDupConsumersAllowSameSubForDifferentTopics) {
     // consumer C should be a different instance from A and B and should be with open state.
     ASSERT_EQ(ResultOk, consumerC.close());
 }
+
+static long regexTestMessagesReceived = 0;
```

Comment:
This gets updated from one thread and read from another. We should use `std::atomic<long>`

Reply:
I can but does it really matter since its only one writer?  Also we doing the same thing in a couple places:
https://github.com/apache/pulsar/blob/master/pulsar-client-cpp/tests/BasicEndToEndTest.cc#L47

Reply:
Yes, the number will be correct because of the single writer, but the reader is technically not guaranteed to see the latest value. Though, yes, with sleep and retries it would settle anyway.